### PR TITLE
Nicolas Carbone python soapbox

### PR DESCRIPTION
**Pull request by nicolascarbone**

### CODE REVIEW

File: soapbox/python/nicolascarbone.md [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+Python does not have access modifiers so use a leading `_` for private methods.
+
+Avoid using leading `__` because that will trigger a process called `name mangling`, it will just put the classname as attribute preffix.
+
+## Coding Practices
+
+As general rule, your code should have high cohesion and low coupling, this will make it easier to maintain, scale, etc.
+
+Some things to keep in mind:
+- Dependency Injection: This will decouple your services from other ones and it will make them easier to test and maintain.
```
Comment (theothermattm) [inline soapbox/python/nicolascarbone.md:72]: Is there a preferred way to do this that you can reference/link?

File: soapbox/python/nicolascarbone.md [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+### Accessing to the database.
+
+Whenever possible use an ORM to access to the database, this can help to make your app compatible with different db engines like:
+- Postgres
+- Mssql
+- Sqlite
+etc
+
+A common option is SqlAlchemy but there are others like peewee.
```
Comment (theothermattm) [inline soapbox/python/nicolascarbone.md:85]: links?

Comment (nicolascarbone) [reply soapbox/python/nicolascarbone.md:85]: good catch !

File: soapbox/python/nicolascarbone.md (hunk 1 above)
Comment (nicolascarbone) [reply soapbox/python/nicolascarbone.md:72]: yes, I'll add links to some implementations

File: soapbox/python/nicolascarbone.md [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+- Use `cls` as first argument to class methods.
+
+### Linting
+
+There are many linters in Python, choose the one you like the most and stick to it.
+Some of them are
+- Pylint
+- Flake8
+- Pylama
+etc
```
Comment (nicolascarbone) [inline soapbox/python/nicolascarbone.md:27]: add links here

File: README.md [hunk 4]
```diff
@@ -199,3 +199,7 @@ These are not meant to be comprehensive _by any means_. You may even find confli
 ## Other Non-Technical Topics
 
 - [Project handoff guide by Simon](soapbox/general/smonn-project-handoff.md)
+
+## Python
```
Comment (theothermattm) [inline README.md:203]: Please add an anchor on the table of contents for this section if you would!

File: soapbox/python/nicolascarbone.md [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+
+    ```
+    cache = mock.Mock()
+    my_service(cache=cache)
+    cache.get.assert_called_with(key)
+    ```
+
+### Frameworks
+
+    <link to fastapi soapbox> ?
```
Comment (theothermattm) [inline soapbox/python/nicolascarbone.md:123]: Maybe leave this out for now?

File: soapbox/python/nicolascarbone.md [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+
+### Virtual Envs
```
Comment (theothermattm) [inline soapbox/python/nicolascarbone.md:125]: I feel like these three sections should be up top... Maybe it's just me but this is the type of stuff I'd be looking for _first_! I always struggle with this crap in python

**Comment by theothermattm** [comment]
Is this still WIP?  Just a couple of minor things to address and I think it's ready to merge personally :)

File: README.md (hunk 4 above)
Comment (nicolascarbone) [reply README.md:202]: I also added an anchor for angular

File: soapbox/python/nicolascarbone.md (hunk 6 above)
Comment (nicolascarbone) [reply soapbox/python/nicolascarbone.md]: LOL!
that makes sense
I moved the environment section to the top

**Comment by nicolascarbone** [comment]
> Is this still WIP? Just a couple of minor things to address and I think it's ready to merge personally :)

I think it is ready now !
do you mind taking another look?

File: README.md [hunk 7]
```diff
@@ -19,6 +19,8 @@
   - [React](#react)
   - [Ruby / Rails](#ruby--rails)
   - [Other Non-Technical Topics](#other-non-technical-topics)
+  - [Angular](#angular)
```
Comment (theothermattm) [inline README.md:22]: Angular is already there and please put Python before react :)

Comment (nicolascarbone) [reply README.md:22]: oh you are right
Python before react just here or also in the soapbox list at the bottom ?

Comment (theothermattm) [reply README.md:22]: Ah, probably both, good catch!

Comment (nicolascarbone) [reply README.md]: @theothermattm done!
I also added a "System dependencies" section

**Review by theothermattm — APPROVED** [review]
This is so good, thanks nico!  I'll let you do the honors of merging when you're ready! :-D

**Pull request merged**